### PR TITLE
Rename the events and function to make it consistent.

### PR DESCRIPTION
**Pull request by deepesh-kn**
Fixes: #477 
- Renamed events and function names as described in #447.
- All existing tests are working.
- No new tests are added in this change.

### CODE REVIEW

File: contracts/gateway/MessageBus.sol [hunk 1]
```diff
@@ -62,7 +62,7 @@ library MessageBus {
     struct Message {
 
         /**
-         * intent hash, this can be staking intent hash, redemption intent
+         * intent hash, this can be stake intent hash, redeem intent
```
Comment (sarvesh-ost) [inline contracts/gateway/MessageBus.sol]: Message bus shouldn't know about stake and redeem probably.

File: contracts/gateway/MockMessageBus.sol [hunk 2]
```diff
@@ -63,7 +63,7 @@ library MockMessageBus {
     struct Message {
 
         /**
-         * intent hash, this can be staking intent hash, redemption intent
+         * intent hash, this can be stake intent hash, redeem intent
```
Comment (sarvesh-ost) [inline contracts/gateway/MockMessageBus.sol]: Same as above comment.

File: contracts/gateway/EIP20CoGateway.sol [hunk 3]
```diff
@@ -90,33 +90,33 @@ contract EIP20CoGateway is CoGateway {
...
         address _redeemer,
         uint256 _redeemerNonce,
         address _beneficiary,
         uint256 _amount
     );
 
-    /** Emitted whenever redemption is completed. */
-    event ProgressedRedemption(
+    /** Emitted whenever redeem is completed. */
+    event ProgressedRedeem(
```
[the remark this answers is not in the document]
Comment (schemar) [reply contracts/gateway/EIP20CoGateway.sol]: I like the suggestion 👍 Is in line with `...Declared`

File: contracts/gateway/EIP20CoGateway.sol [hunk 4]
```diff
@@ -144,7 +144,7 @@ contract EIP20CoGateway is CoGateway {
     /* Struct */
 
     /**
-     * Redeem stores the redemption information about the redeem amount,
+     * Redeem stores the redeem information,
```
Comment (schemar) [inline contracts/gateway/EIP20CoGateway.sol]: Line could be longer before line break

File: contracts/gateway/EIP20CoGateway.sol [hunk 5]
```diff
@@ -226,24 +226,24 @@ contract EIP20CoGateway is CoGateway {
     }
 
     /* External functions */
-
+
```
Comment (schemar) [inline contracts/gateway/EIP20CoGateway.sol]: 😱

File: contracts/gateway/EIP20CoGateway.sol [hunk 6]
```diff
@@ -427,8 +427,8 @@ contract EIP20CoGateway is CoGateway {
         // delete the mint data
         delete mints[_messageHash];
 
-        // Emit RevertStakingIntentConfirmed event
-        emit RevertStakingIntentConfirmed(
+        // Emit RevertStakeIntentConfirmed event
```
Comment (schemar) [inline contracts/gateway/EIP20CoGateway.sol]: Is this comment required? It says exactly the same as the line below it. Further occurrences below.

File: contracts/gateway/EIP20CoGateway.sol [hunk 7]
```diff
@@ -944,16 +944,16 @@ contract EIP20CoGateway is CoGateway {
...
+            // Get the redeem intent hash
+            bytes32 intentHash = GatewayLib.hashRedeemIntent(
+                _amount,
+                _beneficiary,
+                msg.sender,
+                _nonce,
+                _gasPrice,
+                _gasLimit,
+                valueToken
+            );
```
Comment (schemar) [inline contracts/gateway/EIP20CoGateway.sol]: Sure this new indentation is correct?

File: contracts/gateway/EIP20Gateway.sol [hunk 8]
```diff
@@ -530,7 +530,7 @@ contract EIP20Gateway is Gateway {
         stakerNonce_ = message.nonce;
         amount_ = stakes[_messageHash].amount;
 
-        //penalty charged to staker for revert staking
+        // penalty charged to staker for revert stake.
```
Comment (schemar) [inline contracts/gateway/EIP20Gateway.sol]:
```suggestion
        // Penalty charged to staker for revert stake.
```

File: contracts/gateway/GatewayRedeemInterface.sol [hunk 9]
```diff
@@ -1,55 +1,106 @@
...
-        uint256 _blockHeight,
-        bytes calldata _rlpEncodedParentNodes
-    )
-        external
-        returns (bool /*TBD*/);
+	/**
+     * @notice Get the current bounty amount from CoGateway
+     *
+     * @return bounty_ Bounty amount.
+     */
```
Comment (schemar) [inline contracts/gateway/GatewayRedeemInterface.sol]: weird doc indentations (also below)

File: contracts/gateway/GatewayRedeemInterface.sol [hunk 10]
```diff
@@ -1,55 +1,106 @@
...
-
-    function processRevertRedemption(
-        bytes32 _messageHash,
-        uint256 _blockHeight,
-        bytes calldata _rlpEncodedParentNodes
-    )
-        external
-        returns (bool /*TBD*/);
+	/**
+     * @notice Get the current bounty amount from CoGateway
```
Comment (schemar) [inline contracts/gateway/GatewayRedeemInterface.sol]: If these functions are `CoGateway` specific, should they be in a `CoGateway` interface? 🤔

File: contracts/gateway/GatewayRedeemInterface.sol [hunk 11]
```diff
@@ -1,55 +1,106 @@
...
-    )
-        external
-        returns (
-            address redeemer_,
-            bytes32 intentHash_,
-            uint256 nonce_,
-            uint256 gasPrice_
-        );
-
-    function processRevertRedemption(
```
Comment (schemar) [inline contracts/gateway/GatewayRedeemInterface.sol]: is there no equivalent to `processRevertRedemption` anymore? What about `isActivated()`, `redeem`, etc.? Why are those removed as part of this PR?

File: contracts/gateway/MessageBus.sol [hunk 12]
```diff
@@ -61,10 +61,6 @@ library MessageBus {
     /** Message */
     struct Message {
 
-        /**
-         * intent hash, this can be staking intent hash, redemption intent
-         * hash, gateway linking hash
-         */
```
Comment (schemar) [inline contracts/gateway/MessageBus.sol]: No doc at all?

File: contracts/gateway/MockMessageBus.sol [hunk 13]
```diff
@@ -62,10 +62,6 @@ library MockMessageBus {
     /** Message */
     struct Message {
 
-        /**
-         * intent hash, this can be staking intent hash, redemption intent
-         * hash, gateway linking hash
-         */
```
Comment (schemar) [inline contracts/gateway/MockMessageBus.sol]: No doc at all?

File: contracts/gateway/GatewayRedeemInterface.sol (hunk 10 above)
Comment (deepesh-kn) [reply contracts/gateway/GatewayRedeemInterface.sol]: No, these functions are Redeem specific.

File: contracts/gateway/GatewayRedeemInterface.sol (hunk 11 above)
Comment (deepesh-kn) [reply contracts/gateway/GatewayRedeemInterface.sol]: The interface was outdated. `processRevertRedemption` and `isActivated` were not needed.  
`redeem` and `RevertRedeem` are available. It has all the methods that may be required to redeem process

File: contracts/gateway/MessageBus.sol (hunk 12 above)
Comment (deepesh-kn) [reply contracts/gateway/MessageBus.sol]: Added again.

File: contracts/gateway/GatewayRedeemInterface.sol (hunk 9 above)
Comment (deepesh-kn) [reply contracts/gateway/GatewayRedeemInterface.sol]: Fixed.

**Comment by deepesh-kn** [comment]
> Is there a separate ticket for the relevant tests?

We have separate tickets for the test. If we try to cover all tests here then it will be like fixing all the test tickets.

> Why are there changes to package-lock.json if package.json wasn't changed? Sure these changes should be committed?

I did `npm install` on my local, I think package-lock.json is not in git ignore.

File: contracts/gateway/EIP20CoGateway.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (deepesh-kn) [reply contracts/gateway/EIP20CoGateway.sol]: Done.

File: contracts/gateway/GatewayRedeemInterface.sol (hunk 10 above)
Comment (schemar) [reply contracts/gateway/GatewayRedeemInterface.sol]: Then why do the docs say it gets it from CoGateway?

**Comment by schemar** [comment]
> > Is there a separate ticket for the relevant tests?
> 
> We have separate tickets for the test. If we try to cover all tests here then it will be like fixing all the test tickets.
> 
> > Why are there changes to package-lock.json if package.json wasn't changed? Sure these changes should be committed?
> 
> I did `npm install` on my local, I think package-lock.json is not in git ignore.

ok
